pkg/kubectl/resource_printer.go

```diff
@@ -2236,6 +2235,18 @@ func (h *HumanReadablePrinter) PrintObj(obj runtime.Object, output io.Writer) er
 		w = GetNewTabWriter(output)
 		defer w.Flush()
 	}
+
+	// check if the object is unstructured.  If so, let's attempt to convert it to a type we can understand before
+	// trying to print, since the printers are keyed by type.  This is extremely expensive.
```

Comment:
> What is the TODO to make it less expensive?

Clayton wants to change the API server to support printing. We'd use that and this would drop out.